### PR TITLE
`azurerm_resource_provider_registration` - fix test: add timeout for custom importer

### PR DESCRIPTION
**Pull request by teowa**
original failure:
```
------- Stdout: -------
=== RUN   TestAccResourceProviderRegistration_basic
=== PAUSE TestAccResourceProviderRegistration_basic
=== CONT  TestAccResourceProviderRegistration_basic
    testcase.go:113: Step 2/2 error running import: exit status 1
        Error: retrieving Subscription Provider (Subscription: "*******"
        Provider Name: "Microsoft.BlockchainTokens"): the context used must have a deadline attached for polling purposes, but got no deadline
--- FAIL: TestAccResourceProviderRegistration_basic (45.71s)
FAIL
```

test:
```
TF_ACC=1 go test -v ./internal/services/resource -parallel 20 -test.run='TestAccResourceProviderRegistration_' -timeout 1440m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccResourceProviderRegistration_basic
=== PAUSE TestAccResourceProviderRegistration_basic
=== RUN   TestAccResourceProviderRegistration_requiresImport
=== PAUSE TestAccResourceProviderRegistration_requiresImport
=== RUN   TestAccResourceProviderRegistration_feature
=== PAUSE TestAccResourceProviderRegistration_feature
=== CONT  TestAccResourceProviderRegistration_basic
=== CONT  TestAccResourceProviderRegistration_feature
=== CONT  TestAccResourceProviderRegistration_requiresImport
--- PASS: TestAccResourceProviderRegistration_requiresImport (53.87s)
--- PASS: TestAccResourceProviderRegistration_basic (56.66s)
--- PASS: TestAccResourceProviderRegistration_feature (4976.44s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/resource      4976.489s

```

### CODE REVIEW

File: internal/services/resource/resource_provider_registration_resource.go [hunk 1]
```diff
@@ -336,6 +336,9 @@ func (r ResourceProviderRegistrationResource) CustomImporter() sdk.ResourceRunFu
 		client := metadata.Client.Resource.ResourceProvidersClient
 		account := metadata.Client.Account
 
+		ctx, cancel := context.WithDeadline(ctx, time.Now().Add(5*time.Minute))
+		defer cancel()
```
Comment (tombuildsstuff) [inline internal/services/resource/resource_provider_registration_resource.go:340]: it'd be more consistent to make this:

```suggestion
		if _, ok := ctx.Deadline(); !ok {
			var cancelFunc context.CancelFunc
			ctx, cancel = context.WithTimeout(ctx, 5*time.Minute)
			defer cancel()
		}
```

As such, could we update this?

Comment (teowa) [reply internal/services/resource/resource_provider_registration_resource.go]: Thanks, I will change this.

**Comment by teowa** [comment]
```
--- PASS: TestAccResourceProviderRegistration_requiresImport (55.54s)
--- PASS: TestAccResourceProviderRegistration_basic (56.95s)
--- PASS: TestAccResourceProviderRegistration_feature (5430.18s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/resource      5430.214s
```

File: internal/services/resource/resource_provider_registration_resource.go [hunk 2]
```diff
@@ -304,7 +305,28 @@ func (r ResourceProviderRegistrationResource) Delete() sdk.ResourceFunc {
...
+					case Unregistering, Unregistered:
+						existingFeatures = append(existingFeatures, map[string]interface{}{"name": featureName, "registered": false})
+					}
+				}
+				if err := result.NextWithContext(ctx); err != nil {
+					return fmt.Errorf("enumerating features: %+v", err)
+				}
+			}
+
+			err = r.applyFeatures(ctx, metadata, *id, existingFeatures, make([]interface{}, 0))
```
Comment (tombuildsstuff) [inline internal/services/resource/resource_provider_registration_resource.go:329]: #23072 ports this over to using `hashicorp/go-azure-sdk` - can we revert this so this doesn't conflict?

Comment (teowa) [reply internal/services/resource/resource_provider_registration_resource.go:329]: sure

**Comment by tombuildsstuff** [comment]
hey @teowa 

Thanks for pushing that change - @wuxu92 has fixed this in a different manner in #23081, as such whilst I'd like to thank you for this contribution - I'm going to close this in favour of #23081 which has resolved this by passing a context with a timeout to the `import` function.

Thanks!

**Pull request closed without merging**